src/Diagnostics.DataProviders/DataProviders/KustoDataProvider.cs

```diff
@@ -93,6 +101,51 @@ public async Task<DataTable> ExecuteQuery(string query, string stampName, string
             return await _kustoClient.ExecuteQueryAsync(Helpers.MakeQueryCloudAgnostic(_kustoMap, query), _kustoMap.MapCluster(cluster) ?? cluster, _kustoMap.MapDatabase(_configuration.DBName) ?? _configuration.DBName, requestId, operationName);
         }
 
+        public async Task<DataTable> ExecuteQueryOnAllAppAppServiceClusters(string query, string requestId = null, string operationName = null)
+        {
+            if (string.IsNullOrWhiteSpace(operationName))
+            {
+                throw new ArgumentNullException(nameof(operationName), "OperationName cannot be empty. Please supply a name to identify the query.");
+            }
+            if (string.IsNullOrWhiteSpace(query))
+            {
+                throw new ArgumentNullException(nameof(query), "Query cannot be empty. Please supply a query to execute.");
+            }
+            else
+            {
+                List<Task<DataTable>> queryTask = new List<Task<DataTable>>();
+
+                if(IsPublicCloud)
+                {
+                    queryTask.Add(ExecuteQuery(query, "waws-prod-bay-153", null, $"ExecuteQueryOnAllAntaresClusters|{operationName}|WestUS"));
```

Comment:
Hey @nmallick1 , what do you think about iterating over our clusters using the Kusto configuration `KustoClusterNameGroupings` and running the `ExecuteClusterQuery` kind of like this:

```
foreach (var cluster in _configuration.KustoClusterNameGroupings.Split(new char[]{','})){
    queryTask.Add(ExecuteClusterQuery(query, cluster, "wawsprod", null, $"ExecuteQueryOnAllAntaresClusters|{operationName}|{cluster}"))
}
```